source/develop/release-management/features/virt/dedicated-cpus.md

```diff
@@ -0,0 +1,68 @@
+---
+Title: Dedicated CPUs
+Category: feature
+Authors: sabusale
+---
+
+# Dedicated CPUs
+
+## Summary
+
+Once choosing dedicated CPU policy, The guest vCPUs will be strictly pinned to a set of host pCPUs (similarly to static CPU pinning). The set of pCPUs will be chosen to match the required guest CPU topology.
+
+## Owner
+
+*   Name: Saif Abu Saleh (sabusale)
+*   Email: sabusale@redhat.com
+
+
+### Tasks:
+
+
+1- Add dedicated to CPU Pinning Policy enum
+
+2- Scheduling
+
+3- Send vcpupin with cpusets and policy name in VM metadata(only when Policy is dedicated?)
+
+4- Send the policy name from vmStatic policy field
+
+5- Send the cpusets (convert from CurrentCpuPinningField form to array)
+
+6- API: Migrate
+Send cpusets additional parameter (convert from CurrentCpuPinningField form to array)
+
+7- setNumberOfCpus VDSM call
+Send cpusets additional parameter (convert from CurrentCpuPinningField form to array)
+Frontend: Add Dedicated policy dropdown to the UI
+
+
+Steps 3-7: can be done in parallel by different person
+
+
+### Scheduling:
+
+* Find host that have the VM CPU topology on the scheduling phase
+* On run phase, schedule the VM to run on that host
+
+
+###Data Model:
+
+* Save CPU mappings in a field (maybe use CurrentCpuPinningField)?
+* Add used_cpus field to the host to know how many CPUs is used
+
+
+
+#### Example:
+
+1 Host with 8 pCPU
+And all VM’s scheduled to the same host
+
+* VM1 with dedicated CPU with 2 vCPU’s -> used_cpus: 2 current_pinning: 1#1-2#2  
+* VM2 with shared CPU with 2 vCPU’s -> used_cpus: 2 current_pinning null
+* VM1 destroyed -> used_cpus:  used_cpus:0
+
+
+#### Questions:
+* How to get to which actual pCPU number the vCPU’s got pinned?
```

Comment:
You need to decide based on the available pCPUs in the selected host. With the current description you made, you will need to pass on every VM that runs on the host with dedicated policy and start calculating which pCPU are free to use.

Reply:
but how I know to which pCPU the vCPU got allocated on run phase?

Reply:
You create the vCPU to pCPU mapping during the run phase based on the information sent from the host about available CPUs, its topology and the vm's cpu policy.